src/python/pants/util/process_handler.py

```diff
@@ -106,8 +106,8 @@ def join_and_get_output():
 
 def _tee(infile, outfile, return_function):
   accumulator = io.BytesIO()
-  for line in iter(infile.readline, ""):
+  for line in iter(infile.readline, b""):
     accumulator.write(line)
-    outfile.write(line)
+    outfile.buffer.write(line) if PY3 else outfile.write(line)
```

Comment:
In Py2, streams like `stdin` write directly using bytes. In Py3, they write using unicode, and if you want to use bytes you have to go under-the-hood by calling `buffer`